regression/cbmc/havoc_choice/main.c

```diff
@@ -0,0 +1,37 @@
+#include <assert.h>
+#include <stdbool.h>
+
+bool nondet_bool();
+
+int main()
+{
+  char a = 'a';
+  char b = 'b';
+  char c = 'c';
+  char d = 'd';
+
+  char *p =
+    nondet_bool() ? (nondet_bool() ? &a : &b) : (nondet_bool() ? &c : &d);
+
+  __CPROVER_havoc_object(p);
+
+  if(p == &a)
+    assert(a == 'a');
+  else
+    assert(a == 'a');
```

Comment:
❓ Why the `then` and `else` branches of the `if` are identical?

Similar below

Reply:
We want to see the `SUCCESS` or `FAILURE` depending on the condition, see the `.desc` file below.